src/stargate/modules/nft/clients/queryClient.ts

```diff
@@ -0,0 +1,100 @@
+
+import { HttpEndpoint, Tendermint34Client } from "@cosmjs/tendermint-rpc";
+import { QueryClient, StargateClient, StargateClientOptions } from "@cosmjs/stargate";
+import { NftExtension, setupNftExtension } from "../../../modules/nft/queries";
+import { QueryApprovalsIsApprovedForAllResponse, QueryApprovalsNFTResponse, QueryCollectionResponse, QueryDenomByNameRequest, QueryDenomByNameResponse, QueryDenomBySymbolResponse, QueryDenomRequest, QueryDenomResponse, QueryDenomsResponse, QueryNFTResponse, QueryOwnerResponse, QuerySupplyResponse } from "../../../modules/nft/proto-types/query";
+import { checkValidNftDenomId, checkValidAddress } from "../../../../utils/checks";
+import { PageRequest } from "cosmjs-types/cosmos/base/query/v1beta1/pagination";
+
+export class NFTQueryClient{
+    private readonly nftQueryClient: QueryClient & NftExtension;
+    
+
+    constructor(tmClient:Tendermint34Client) {
+        this.nftQueryClient = QueryClient.withExtensions(tmClient,setupNftExtension)
+    }
+
+    public async getNftDenomSupply(denomId: string, owner?: string): Promise<QuerySupplyResponse> {
+        checkValidNftDenomId(denomId);
+
+        if (owner !== undefined) {
+            checkValidAddress(owner);
```

Comment:
would't it be better if this check is directly into the `checkValidAddress` function?

Reply:
Actually, this check is specific to this particular method , and not the `checkValidAddress ` function itself.  Reason is that here  the `owner` parameter is optional and the validity check should be conducted only if such a parameter has been provided. `checkValidAddress` is also used in multiple methods where such a check is not needed. A cleaner way for this method might be to assign a default value to `owner` parameter, but this would change the method API a bit (because of parameter reordering), which I think is unnecessary.